subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/projectmodule/DefaultProjectDependencyPublicationResolver.java

```diff
@@ -79,17 +79,19 @@ public <T> T resolve(Class<T> coordsType, ProjectInternal project) {
         // Select all entry points. An entry point is a publication that does not contain a component whose parent is also published
         Set<SoftwareComponent> ignored = new HashSet<>();
         for (ProjectComponentPublication publication : publications) {
-            if (publication.getComponent() != null && publication.getComponent() instanceof ComponentWithVariants) {
-                ComponentWithVariants parent = (ComponentWithVariants) publication.getComponent();
+            SoftwareComponent component = publication.getComponent().getOrNull();
+            if (component instanceof ComponentWithVariants) {
+                ComponentWithVariants parent = (ComponentWithVariants) component;
                 ignored.addAll(parent.getVariants());
             }
         }
         Set<ProjectComponentPublication> topLevel = new LinkedHashSet<>();
         Set<ProjectComponentPublication> topLevelWithComponent = new LinkedHashSet<>();
         for (ProjectComponentPublication publication : publications) {
-            if (!publication.isAlias() && (publication.getComponent() == null || !ignored.contains(publication.getComponent()))) {
+            SoftwareComponent component = publication.getComponent().getOrNull();
+            if (!publication.isAlias() && (component == null || !ignored.contains(component))) {
```

Comment:
🧐 Is the null check needed here?  `ignored` won't ever contain `null`.

Reply:
Removed